packages/common/src/content/_internal.ts

```diff
@@ -89,8 +82,177 @@ const MAX_SIZE = 5000000;
+      // but instead of showing the page's metadata on /documents/about
+      // but we render the page on the pages route
+      path = "/pages";
+    } else if (familiesWithPluralizedRoute.indexOf(family) > -1) {
+      // the rule: route name is plural of family name
+      path = `/${family}s`;
+    }
+    contentUrl = `${path}/${identifier}`;
+  }
+  return contentUrl;
+};
+
+export const isPageType = (type: string) =>
+  ["Hub Page", "Site Page"].includes(type);
+
+const getSolutionUrl = (
+  type: string,
+  identifier: string,
+  typeKeywords?: string[]
+): string => {
+  let hubUrl;
+  if (type === "Solution") {
+    // NOTE: as per the above spreadsheet,
+    // solution types are now in the Template family
+    // but we don't send them to the route for initiative templates
+    if (typeKeywords?.indexOf("Deployed") > 0) {
+      // deployed solutions go to the generic content route
+      hubUrl = `/content/${identifier}`;
+    }
+    // others go to the solution about route
+    hubUrl = `/templates/${identifier}/about`;
+  }
+  return hubUrl;
+};
+
+// metadata
+export interface IMetadataPaths {
+  updateFrequency: string;
+  metadataUpdateFrequency: string;
+  metadataUpdatedDate: string;
+  reviseDate: string;
+  pubDate: string;
+  createDate: string;
+}
+
+export function getMetadataPath(identifier: keyof IMetadataPaths) {
+  // NOTE: i have verified that this will work regardless of the "Metadata Style" set on the org
+  const metadataPaths: IMetadataPaths = {
+    updateFrequency:
+      "metadata.dataIdInfo.resMaint.maintFreq.MaintFreqCd.@_value",
+    reviseDate: "metadata.dataIdInfo.idCitation.date.reviseDate",
+    pubDate: "metadata.dataIdInfo.idCitation.date.pubDate",
+    createDate: "metadata.dataIdInfo.idCitation.date.createDate",
+    metadataUpdateFrequency: "metadata.mdMaint.maintFreq.MaintFreqCd.@_value",
+    metadataUpdatedDate: "metadata.mdDateSt",
+  };
+  return metadataPaths[identifier];
+}
+
+export function getValueFromMetadata(
+  metadata: any,
+  identifier: keyof IMetadataPaths
+) {
+  const path = getMetadataPath(identifier);
+  return path && getProp(metadata, path);
+}
+
+export enum DatePrecision {
+  Year = "year",
+  Month = "month",
+  Day = "day",
+  Time = "time",
+}
+
+/**
+ * Parses an ISO8601 date string into a date and a precision.
+ * This is because a) if somone entered 2018, we want to respect that and not treat it as the same as 2018-01-01
+ * and b) you cannot naively call new Date with an ISO 8601 string that does not include time information
+ * For example, when I, here in mountain time, do new Date('2018').getFullYear() I get "2017".
+ * This is because when you do not provide time or timezone info, UTC is assumed, so new Date('2018') is 2018-01-01T00:00:00 in UTC
+ * which is actually 7 hours earlier here in mountain time.
+ *
+ * @param {string} isoString
+ * @return { date: Date, precision: DatePrecision }
+ */
+export function parseISODateString(isoString: string) {
```

Comment:
Does Date.parse() work?

Reply:
I generally try to [avoid](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) `Date.parse()`. But also see the comment above. We're trying to extract not just a date but also the level of precision the user intended.